quinn/src/endpoint.rs

```diff
@@ -191,10 +196,19 @@ impl EndpointInner {
                 Ok(Async::Ready((n, addr, ecn))) => {
                     match self.inner.handle(now, addr, ecn, (&buf[0..n]).into()) {
                         Some((handle, DatagramEvent::NewConnection(conn))) => {
-                            let conn = self.create_connection(handle, conn);
-                            self.incoming.push_back(ConnectionDriver(conn));
-                            if let Some(task) = self.blocked_incoming.take() {
-                                task.notify();
+                            let conn = ConnectionDriver(self.create_connection(handle, conn));
+                            if let Some(ref mut incoming) = self.incoming {
+                                incoming.push_back(conn);
+                                if let Some(task) = self.blocked_incoming.take() {
+                                    task.notify();
+                                }
+                            } else {
```

Comment:
Maybe add a method on `ConnectionDriver` to do the dirty here, so that `ConnectionInner::inner` doesn't have to be `pub(crate)`?